source/common/http/header_map_impl.cc

```diff
@@ -616,6 +616,7 @@ HeaderMapImpl::HeaderEntryImpl& HeaderMapImpl::maybeCreateInline(HeaderEntryImpl
     return **entry;
   }
 
+  addSize(key.get().size());
```

Comment:
Was this a previous bug?

Reply:
Ah. Nope, there was no `addSize` here because the method invalidated the cached size right above this.
I should remove that reset. It was there because before this PR the only caller of this method was `insert##name()` which also has resets byte size. Now there are `HeaderMapImpl` methods that are safe for tracking. I'll add in a test case into `header_map_impl_test` for this.

Reply:
OK got it.